File: foundation/foundation-messaging/foundation-messaging-python/aissemble-foundation-messaging-python-client/src/aissemble_messaging/exception/topic_not_supported_error.py
###
# #%L
# aiSSEMBLE::Foundation::Messaging Python
# %%
# Copyright (C) 2021 Booz Allen
# %%
# This software package is licensed under the Booz Allen Public License. All Rights Reserved.
# #L%
###
from aissemble_messaging.exception.base_messaging_error import BaseMessagingError


class TopicNotSupportedError(BaseMessagingError):
    """
    An error thrown when the caller has attempted to subscribe to a topic that cannot be found in the service.
    """

    topic: str

    def __init__(self, topic: str):
        self.topic = topic
        self.message = "Could not find a topic to subscribe to named " + topic

File: foundation/foundation-messaging/foundation-messaging-python/foundation-messaging-python-service/src/main/java/com/boozallen/aissemble/messaging/python/exception/TopicNotSupportedError.java
package com.boozallen.aissemble.messaging.python.exception;

/*-
 * #%L
 * aiSSEMBLE::Foundation::Messaging::Python::Service
 * %%
 * Copyright (C) 2021 Booz Allen
 * %%
 * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
 * #L%
 */

public class TopicNotSupportedError extends Exception {

    private final String topic;

    public TopicNotSupportedError(String topicName) {
        super("Could not find a topic to subscribe to named " + topicName);
        topic = topicName;
    }
}
